sdks/java/core/src/test/java/org/apache/beam/sdk/io/AvroIOTransformTest.java

```diff
@@ -247,41 +248,59 @@ public void testRead() throws Exception {
 
     private static final String WRITE_TRANSFORM_NAME = "AvroIO.Write";
 
-    private List<AvroGeneratedUser> readAvroFile(final File avroFile) throws IOException {
-      final DatumReader<AvroGeneratedUser> userDatumReader =
-          new SpecificDatumReader<>(AvroGeneratedUser.class);
-      final List<AvroGeneratedUser> users = new ArrayList<>();
-      try (DataFileReader<AvroGeneratedUser> dataFileReader =
-          new DataFileReader<>(avroFile, userDatumReader)) {
-        while (dataFileReader.hasNext()) {
-          users.add(dataFileReader.next());
+    private <T> List<T> readAvroFile(final File avroFile, boolean generic) throws IOException {
+      if (generic) {
+        final DatumReader<GenericRecord> datumReader = new GenericDatumReader<>(SCHEMA);
```

Comment:
Looks like everything except lines 253 vs. 263 is the same and should probably be deduplicated.

Reply:
of course! thanks for pointing